tests/sources.cpp

```diff
@@ -300,54 +300,50 @@ BOOST_AUTO_TEST_CASE( fivoxSpikes_stream_source_frame_range )
     uri.setScheme( _monsteerPluginScheme );
 
     fivox::URIHandler params(
-        "fivoxspikes://?dt=0.1&duration=1&spikes=" + std::to_string( uri ));
-    auto filter = params.newImageSource< float >();
+        fivox::URI( "fivoxspikes://?dt=0.1&duration=1&spikes=" +
+                    std::to_string( uri )));
+    auto filter = params.newImageSource< fivox::FloatVolume >();
     fivox::EventSourcePtr source = filter->getEventSource();
 
     lunchbox::sleep( STARTUP_DELAY );
 
     brion::Spikes spikes;
     for( uint32_t i = 0; i <= 50; ++i )
-        spikes.insert( std::make_pair( i / 100.0f, i ));
-    spikeWriter.writeSpikes( spikes );
+        spikes.push_back( std::make_pair( i / 100.0f, i ));
+    spikeWriter.write( spikes );
 
     // Time range: [0, 0.5](ms)
     // Since duration=1(ms), no full frames are available: [0,0)
     BOOST_CHECK_EQUAL( source->getFrameRange(), fivox::Vector2ui( 0, 0 ));
 
     spikes.clear();
     for( uint32_t i = 51; i <= 100; ++i )
-        spikes.insert( std::make_pair( i / 100.0f, i ));
-    spikeWriter.writeSpikes( spikes );
+        spikes.push_back( std::make_pair( i / 100.0f, i ));
+    spikeWriter.write( spikes );
 
     lunchbox::sleep( WRITE_DELAY );
-
-    // The very last spike is not digested by the internal SpikeReportReader
-    // because to provide complete data inside a time window it cannot advance
-    // past time t_max until one spike with t > t_max arrives.
     BOOST_CHECK_EQUAL( source->getFrameRange(), fivox::Vector2ui( 0, 0 ));
 
     spikes.clear();
     for( uint32_t i = 101; i <= 120; ++i )
-        spikes.insert( std::make_pair( i / 100.0f, i ));
-    spikeWriter.writeSpikes( spikes );
+        spikes.push_back( std::make_pair( i / 100.0f, i ));
+    spikeWriter.write( spikes );
 
     lunchbox::sleep( WRITE_DELAY );
-    BOOST_CHECK_EQUAL( source->getFrameRange(), fivox::Vector2ui( 0, 1 ));
+    BOOST_CHECK_EQUAL( source->getFrameRange(), fivox::Vector2ui( 0, 2 ));
```

Comment:
Due to the changes in the low level spike report specification (which is more strict now), the way the sender and receiver get in synch it's slightly different. Not that before was wrong, but I actually prefer how it works right now because the current code doesn't require the sender to move past the end of a time window to let the receiver know that the frame is complete. This allows the receiver to be more up to date with the sender and it's the reason for the changes in this test.